msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackGenerator.java

```diff
@@ -513,6 +513,12 @@ private void writeByteArrayTextKey(byte[] text, int offset, int len) throws IOEx
         addValueNode(new String(text, offset, len, DEFAULT_CHARSET));
     }
 
+    @Override
+    public void writeFieldId(long id) throws IOException
+    {
+        addKeyNode(id);
```

Comment:
@komamitsu @xerial For backwards compatibility, I imagine we need some sort of feature flag that defaults to the prior implementation. Do you guys have any suggestions for how that flag should be defined? Should I just use the [`MessagePack.PackerConfig`](https://github.com/msgpack/msgpack-java/blob/59219838ffb6236eca84601dad1dd267541c070b/msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackGenerator.java#L52), or should I pass it [directly from the `MessagePackFactory` to the `MessagePackGenerator`](https://github.com/msgpack/msgpack-java/blob/59219838ffb6236eca84601dad1dd267541c070b/msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackFactory.java#L87)?
```suggestion
        if (writeIntegerMapKeysAsStringKeys) {
            super.writeFieldId(id);
        } else {
            addKeyNode(id);
        }
```

Reply:
Sorry, for the delayed reply. `PackerConfig` is for the `core` configuration and shouldn't be used for `jackson-dataformat-msgpack`. I think it's okay to pass the flag like `reuseResource`.